### PR TITLE
[PLAT-5055] Support macOS E2E tests

### PR DESCRIPTION
**Pull request by Cawllec**
## Goal

Part 1 of the addition of MacOS e2e tests to CI by @nickdowell

Enables standard e2e test features to be run on MacOS through an appium server.

Tests that won't run in this format are tagged with `@skip_macos`, and will be skipped using `--tags='not @skip_macos'` in test invocations.

Part 2 will be a basic WIP pipeline running these tests using our existing infrastructure

### CODE REVIEW

**Comment by Cawllec** [comment]
@nickdowell I realise you made most of the initial changes, so if you could just do a feedback review and I'll wait for @twometresteve for approval

**Review by nickdowell — APPROVED** [review]
I have successfully run the test suite locally :tada:

File: features/app_and_device_attributes.feature
Comment (twometresteve) [inline features/app_and_device_attributes.feature:23]: Do we have a ticket to cover things like this, or do we need to put an iOS specific step in before we can merge this?

File: features/barebone_tests.feature
Comment (twometresteve) [inline features/barebone_tests.feature:187]: Same question here - I don't think we would want the iOS tests to take a step back from this change.

File: features/crashprobe.feature
Comment (twometresteve) [inline features/crashprobe.feature:37]: What was the outcome here?  I'm happy to raise tickets for follow-on work, so long as we reference the ticket numbers.

File: features/fixtures/shared/scenarios/Scenario.m
Comment (twometresteve) [inline features/fixtures/shared/scenarios/Scenario.m:77]: Please reference PLAT-5827, which I've raised for MazeRunner to provide a suitable endpoint.  I've made a note of that ticket to reinstate this line.

File: features/handled_errors.feature
Comment (twometresteve) [inline features/handled_errors.feature:65]: What's the right answer here?  Follow-on ticket?

File: features/steps/ios_steps.rb
Comment (twometresteve) [inline features/steps/ios_steps.rb:33]: nit: single quotes

File: features/steps/ios_steps.rb
Comment (twometresteve) [inline features/steps/ios_steps.rb:54]: Please can we keep this comment under the `else`.

File: features/steps/ios_steps.rb
Comment (twometresteve) [inline features/steps/ios_steps.rb:33]: I think checking `Maze.config.os` (== 'macos') would be preferable as it's explicitly passed on the CLI and not a product of the Appium connection. (Applies in various other places too)

File: features/unhandled_signal.feature
Comment (twometresteve) [inline features/unhandled_signal.feature:24]: Follow on ticket?

**Review by twometresteve — COMMENTED** [review]
Some comments raised for little improvements and on how we handle the follow on work - but looking really good.

File: features/app_and_device_attributes.feature
Comment (Cawllec) [reply features/app_and_device_attributes.feature:23]: This will be resolved by adding a nullable tag to the platform-dependent test assertion steps.

File: features/barebone_tests.feature
Comment (Cawllec) [reply features/barebone_tests.feature:187]: This will be resolved by adding a nullable tag to the platform-dependent test assertion steps.

File: features/crashprobe.feature
Comment (Cawllec) [reply features/crashprobe.feature:37]: @nickdowell do you have any info on this?  Would you be able to create a ticket and I'll reference it here?

File: features/unhandled_signal.feature
Comment (Cawllec) [reply features/unhandled_signal.feature:24]: @nickdowell Is there an existing ticket to address this?  I can create one otherwise.

Comment (nickdowell) [reply features/unhandled_signal.feature:24]: I don't this there is a resolution to this, other than maybe not pedantically checking the stack traces in scenarios?

File: features/handled_errors.feature
Comment (Cawllec) [reply features/handled_errors.feature:65]: This will be resolved by adding a nullable tag to the platform-dependent test assertion steps.

File: features/crashprobe.feature
Comment (nickdowell) [reply features/crashprobe.feature:37]: Afraid I don't remember the details of this one; have raised ticket PLAT-5860 since it seems worth further investigation.

File: features/unhandled_signal.feature
Comment (nickdowell) [reply features/unhandled_signal.feature:24]: 👉 PLAT-5861

**Review by twometresteve — APPROVED** [review]
Changes LGTM if @nickdowell is happy with how it functions.

**Review by nickdowell — APPROVED** [review]
```
136 scenarios (136 passed)
1409 steps (1409 passed)
18m30.918s
+++ All scenarios complete
```

:tada:

**Pull request merged**
